insights/parsers/samba.py

```diff
@@ -108,3 +109,31 @@ def parse_content(self, content):
             # Merge same-named sections just as samba's `testparm` does.
             new_dict[new_key].update(old_section)
         self.data._sections = new_dict
+
+
+add_filter(Specs.testparm_v_s, ["["])
+add_filter(Specs.testparm_v_s, ["Server role:"])
+
+
+@parser(Specs.testparm_v_s)
+class SambaConfigs(SambaConfig):
+    """
+    This parser reads the Samba configuration from command `testparm -v -c` which is more
+    reliable than parsing the config as well as contains server role.
+    """
+
+    def __init__(self, *args, **kwargs):
+        self.server_role = None
```

Comment:
Please add the corresponding documentation for this new attribute. e.g. to the "Attributes:" docstring of this class.


BTW, since the `parse_content` will also be called by the `__init__` of the base class, this `self.server_role` can also be declared in the `parse_content` (so the re-implentmentation of this `__init__` can be saved :-)

Reply:
You are right. Liner in PyCharm complains about unknown attributes, so that is why I usually add it in `__init__` but I agree it is unnecessary.